scripts/copyFilesUtils.mjs

```diff
@@ -202,6 +202,10 @@ export async function createPackageFile(useEsmExports = false) {
     newPackageData.types = './index.d.ts';
   }
 
+  if (newPackageData.publishConfig?.directory) {
+    delete newPackageData.publishConfig.directory;
```

Comment:
Yep, is what I was also doing in https://github.com/mui/material-ui/pull/46467/files. 🤔  curious that that didn't work for me when I tried the same in X. I must have been looking at the wrong output or something.